### PR TITLE
[Chore + Mini-Features] Metis exercise/lecture discussion section improvements

### PR DESCRIPTION
**Pull request by lschlesinger**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

The discussion section (former "Postings" former "Q&A") on exercise and lecture pages had some improper translations, such as "Postings" (in German and English), which I wanted to change. Thereby, I realized that many other styling issues should be resolved, the structure in general could be improved to a cleaner version and some mini-features should be added. 
I decided to go for a "Chore+Mini-Features PR" 😄 

### Description

**Styling and Appearance**
- change of colors (too many nested grey boxes, icons too black), sizes of icons
- introduce different icons for post (bubble) and answer post (two bubbles)
- fixed button misalignment in edit mode (cancel and save button)
- re-structuring of answers and posts e.g. line divider between posts

**Translation and labels**
- remove labels and tooltips where not necessary (e.g. post, answer above the content -> label, style and placement should be enough to identify a posts and answers)
- section name is changed to `Discussion`
- German translation for Post is `Beitrag`
- all Metis related translations for the discussion section and overview were factored out in dedicated translation files
- adapt overlooked translation keys for course and exercise statistics

**Refactoring/Renaming notification domain**
The notification component was not refactored yet (server side).

**Mini-Features**
- Input validation: it should not be possible to created or save (after edit) posts or answers without content
- Loading feedback: the save-button was changed to a button with spinner indication during loading time (It happened to me sometimes, that I got impatient when I did not receive any feedback of the loading process and clicked on save two times, which led to duplicated posts)

### Steps for Testing

Note: I did not change the logic of voting or approving, please focus on reporting issues regarding the style or the wording of the discussion section, or bug related to the mini-features mentioned above.

1. Log in to Artemis
2. Navigate to an Exercise or Lecture page and use the discussion section
3. As student: add posts and answers, edit own posts
4. As instructor: add posts and answers (automatically approved), edit posts and answers of anybody
5. Check if the notifications (bell at the top) for the new actions are created properly (with the new terms: post/Beitrag, reply/Antwort)

As instructor:
1. Check course statistics: no missing translation keys for post related statistics
2. Check exercise statistics: no missing translation keys for post related statistics

### Screenshots

Before:
Bildschirmfoto 2021-06-13 um 22 17 20
After:
Bildschirmfoto 2021-06-13 um 22 15 29

### CODE REVIEW

**Review by FuchsDominik — COMMENTED** [review]
Code looks good, could not test yet since the build does not seem to start. I personally liked the color-wise distinction of the post and the background but new one is fine too :) Rest looks really good for what i saw in the screenshots

**Review by MDK5 — CHANGES_REQUESTED** [review]
Tested in testing session on ts1. The styling is a bit broken, if the problem statement is rather wide, the posting section gets moved to the bottom.
grafik

**Comment by lschlesinger** [comment]
> Tested in testing session on ts1. The styling is a bit broken, if the problem statement is rather wide, the posting section gets moved to the bottom.

Should be fixed with a4ce7a3

File: src/main/webapp/app/overview/postings/postings.component.ts [hunk 1]
```diff
@@ -130,6 +132,7 @@ export class PostingsComponent implements OnInit, AfterViewInit {
             this.posts.push(postResponse.body!);
             this.postContent = undefined;
             this.isEditMode = false;
+            this.isLoading = false;
         });
```
[the remark this answers is not in the document]
Comment (lschlesinger) [reply src/main/webapp/app/overview/postings/postings.component.ts]: I updated the logic in 846b040, `isLoading` is now always set to false on completion (no matter if failed or succeeded)

[the remark this answers is not in the document]
Comment (lschlesinger) [reply src/main/webapp/app/overview/postings/postings.component.ts]: and btw. the load-buttons that looked a bit odd (with too much space to the left) are fixed as well :)

**Review by willergeorg — APPROVED** [review]
Tested on TS2. Worked for me.

**Review by NicolasUllmann — DISMISSED** [review]
Tested on TS1. Everything worked as expected.

**Comment by lschlesinger** [comment]
I had to commit a fix for the overflow problem. If a UML diagram was rendered in the problem statement of a programming exercise and the screen size was decreased, the behavior of the discussion section was not as intended.
@Reviwers, please check out this scenario and report, if the section is moved below the problem statement as spans over the full page width.

File: src/main/webapp/app/exercises/programming/shared/instructions-render/extensions/programming-exercise-plant-uml.extension.ts [hunk 2]
```diff
@@ -45,6 +45,9 @@ export class ProgrammingExercisePlantUmlExtensionWrapper implements ArtemisShowd
                 tap((plantUmlSvg: string) => {
                     const plantUmlHtmlContainer = document.getElementById(`plantUml-${index}`);
                     if (plantUmlHtmlContainer) {
+                        // We add a max-width: 100% style attribute to the svg element
+                        // to prevent overflowing the enclosing container
+                        plantUmlSvg = plantUmlSvg.replace(`style="width`, `style="max-width:100%;width`);
```
Comment (lschlesinger) [inline src/main/webapp/app/exercises/programming/shared/instructions-render/extensions/programming-exercise-plant-uml.extension.ts:50]: @FrankeLukas hope this is an allowed fixed for the overflow problem on an exercise page with rendered uml model (as svg). It was not possibly for me to solve the problem without restricting the svg width to max of 100%.

**Review by KiraNickel — CHANGES_REQUESTED** [review]
Tested on ts3. For me, in both languages and both as student and instructor the text for most options is not displayed correctly (see screenshots)

English and German Section with no posts:
grafik

All other boxes in both languages, like answering/editing/deleting/adding new post.
grafik

**Comment by lschlesinger** [comment]
> Tested on ts3. For me, in both languages and both as student and instructor the text for most options is not displayed correctly (see screenshots)
@KiraNickel that's most probably a caching issue. Could you please try to empty your browser cache and reload the page before testing it once again?

File: src/main/webapp/i18n/en/instructorCourseDashboard.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
             "action": "Action",
             "exerciseDashboard": "Exercise dashboard",
             "rating": "Ratings",
-            "questionsAndAnswers": "Questions",
+            "discussion": "Disucssions",
```
Comment (ivanchimeno) [inline src/main/webapp/i18n/en/instructorCourseDashboard.json:26]:
```suggestion
            "discussion": "Discussions",
```

Comment (lschlesinger) [reply src/main/webapp/i18n/en/instructorCourseDashboard.json]: damn! solved 6627496

**Review by MDK5 — APPROVED** [review]
Tested in testing session on ts5, worked well

**Review by FuchsDominik — APPROVED** [review]
Couldn't find any missing translations or other problems during testing

**Review by canyelten — DISMISSED** [review]
Tested on TS2, works as expected.

**Review by KiraNickel — DISMISSED** [review]
Tested on ts3, works as expected 👍🏼  (sorry I forgot to clear the cache before)

**Review by ivanchimeno — DISMISSED** [review]
Tested and works!

**Review by ArtemTsurikov — DISMISSED** [review]
Tested on TS1, everything works.

**Pull request merged**